pinot-core/src/main/java/org/apache/pinot/core/util/ListenerConfigUtil.java

```diff
@@ -143,6 +143,27 @@ private ListenerConfigUtil() {
     return listeners;
   }
 
+  public static List<ListenerConfig> buildMinionAdminConfigs(PinotConfiguration minionConf) {
+    List<ListenerConfig> listeners = new ArrayList<>();
+
+    String adminApiPortString = minionConf.getProperty(CommonConstants.Minion.CONFIG_OF_ADMIN_API_PORT);
+    if (adminApiPortString != null) {
+      listeners.add(new ListenerConfig(CommonConstants.HTTP_PROTOCOL, DEFAULT_HOST, Integer.parseInt(adminApiPortString),
+          CommonConstants.HTTP_PROTOCOL, new TlsConfig()));
+    }
+
+    TlsConfig tlsDefaults = TlsUtils.extractTlsConfig(minionConf, CommonConstants.Minion.MINION_TLS_PREFIX);
+    listeners.addAll(buildListenerConfigs(minionConf, "pinot.minion.adminapi", tlsDefaults));
+
+    // support legacy behavior < 0.7.0
+    if (listeners.isEmpty()) {
```

Comment:
I'd kill this part. definitely no need for legacy support here since this is a new feature

Reply:
Yeah, was thinking the same, but thought it is "virtual" legacy as folks may use the same convention as other components, otherwise, this one becomes odd one out, and hard to explain to users.

Reply:
We'll have move to ListenerSpecs anyways. Otherwise, SSL will forever be off by default and any pinot user will be non-compliant with the GDPRs (and knock-offs)

listener spec examples: https://docs.pinot.apache.org/operators/tutorials/configuring-tls-ssl